### ISSUE TITLE
Homepage shows connections header even without given permissions

### ISSUE THREAD

**Issue by david-kn** [nautobot/nautobot#4281]
### Environment
* Nautobot version (Docker tag too if applicable): 1.6.0
* Python version:
* Database platform, version:
* Middleware(s):

### Steps to Reproduce
1. Limit user's permissions only to `view` a model `Devices` in DCIM.

### Expected Behavior
Only `Devices` should be displayed within the DCIM section on a homepage.

### Observed Behavior
`Connections` header is visible as well on the homepage.
![image](https://github.com/nautobot/nautobot/assets/14314517/0af2f900-6f37-44e8-a999-9954e0bcd1e6)

**Comment by jathanism** [comment]
Confirmed that this does in fact still show. However as this is a `HomePageGroup` that contains other permission-constrained objects. We likely want to improve such groupings to only display if they have member items that are eligible for display based on permissions.

**Comment by david-kn** [comment]
Yeah, that sounds good. I know it's not really a big bug or much of a problem, but still I thought you would want to know about it :-)
